api/apps/geoprocessing/src/modules/scenario-planning-units-inclusion/scenario-planning-units-inclusion-processor.ts

```diff
@@ -121,9 +121,7 @@ export class ScenarioPlanningUnitsInclusionProcessor
 
     if (
       new Set(
-        [...uniquePuIdsToInclude].filter((i) =>
-          uniquePuIdsToExclude.has(i),
-        ),
+        [...uniquePuIdsToInclude].filter((i) => uniquePuIdsToExclude.has(i)),
```

Comment:
The "algorithm" started to be a bit complicated, thus I would recommend extracting the selection based on inputs  (4x arrays) to standalone class with named methods so that future us could get what is going us a bit faster.

Reply:
I agree - I decided to keep things here - I think that the use of lodash set operators as advised by Dominik and the few comments in the source could do... given the circumstances, at least 😱